packages/arb-token-bridge-ui/src/hooks/useArbTokenBridge.ts

```diff
@@ -164,60 +169,89 @@ export const useArbTokenBridge = (
   const depositEth = async ({
     amount,
     l1Signer,
-    txLifecycle
+    txLifecycle,
+    destinationAddress
   }: {
     amount: BigNumber
     l1Signer: Signer
-    txLifecycle?: L1EthDepositTransactionLifecycle
+    txLifecycle?:
+      | L1EthDepositTransactionLifecycle & L1ContractCallTransactionLifecycle
+    destinationAddress?: string
   }) => {
     const ethBridger = await EthBridger.fromProvider(l2.provider)
 
-    let tx: L1EthDepositTransaction
+    let tx: L1EthDepositTransaction & L1ContractCallTransaction
+    let receipt:
+      | L1EthDepositTransactionReceipt
+      | L1ContractCallTransactionReceipt
+
+    const senderAddress = await l1Signer.getAddress()
 
     try {
-      tx = await ethBridger.deposit({
-        amount,
-        l1Signer
-      })
+      tx = (
+        destinationAddress && destinationAddress !== senderAddress
+          ? // send ETH to a custom address
+            await ethBridger.depositTo({
+              amount,
+              l1Signer,
+              l2Provider: l2.provider,
+              destinationAddress
+            })
+          : await ethBridger.deposit({
+              amount,
+              l1Signer
+            })
+      ) as typeof tx
 
       if (txLifecycle?.onTxSubmit) {
         txLifecycle.onTxSubmit(tx)
       }
+      addTransaction({
+        type: 'deposit-l1',
+        status: 'pending',
+        value: utils.formatEther(amount),
+        txID: tx.hash,
+        assetName: 'ETH',
+        assetType: AssetType.ETH,
+        sender: walletAddress,
+        l1NetworkID,
+        l2NetworkID
+      })
+      receipt = await tx.wait()
+      if (txLifecycle?.onTxConfirm) {
+        txLifecycle.onTxConfirm(receipt)
+      }
     } catch (error: any) {
       if (txLifecycle?.onTxError) {
         txLifecycle.onTxError(error)
       }
       return error.message
     }
 
-    addTransaction({
-      type: 'deposit-l1',
-      status: 'pending',
-      value: utils.formatEther(amount),
-      txID: tx.hash,
-      assetName: 'ETH',
-      assetType: AssetType.ETH,
-      sender: walletAddress,
-      l1NetworkID,
-      l2NetworkID
-    })
-
-    const receipt = await tx.wait()
-
-    if (txLifecycle?.onTxConfirm) {
-      txLifecycle.onTxConfirm(receipt)
+    let retryableCreationTxID = ''
+    if (destinationAddress) {
```

Comment:
We can optimize the next few lines of trying to get `retryableCreationTxID`. Currently even if it's a SC/custom destination transfer we still fetch `const [ethDepositMessage] = await receipt.getEthDeposits(l2.provider)` which is not required. It should be an `if-else` case ideally.

This will also get rid of the need of having an empty return in both cases... We can simply have `if(!retryableCreationTxID) return;`